lib/template/hil.go

```diff
@@ -0,0 +1,51 @@
+package template
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/hil"
+	"github.com/hashicorp/hil/ast"
+)
+
+// InterpolateHIL processes the string as if it were HIL and interpolates only
+// the provided string->string map as possible variables.
+func InterpolateHIL(s string, vars map[string]string, lowercase bool) (string, error) {
```

Comment:
Do you plan to use this helper in the existing AuthMethod code too? Presumably we already have something similar there?

Reply:
Yep, just updated and removed the helper from agent/consul/util.go